File: Day41/Part4/string_reduction_double_rule.py
def can_reduce_string(s: str) -> bool:
    n = len(s)
    if n == 0:
        return True
    return n != 1

if __name__ == "__main__":
    print(can_reduce_string("abcaacb"))  # True
    print(can_reduce_string("a"))        # False
    print(can_reduce_string("aa"))       # True

File: Day41/Part4/string_reduction_double_rule.cpp
#include <bits/stdc++.h>
using namespace std;

bool can_reduce_string(const string &s){
    int n = (int)s.size();
    if(n==0) return true;
    return n != 1;
}

int main(){
    cout << (can_reduce_string("abcaacb")?"true":"false") << '\n';
    cout << (can_reduce_string("a")?"true":"false") << '\n';
    cout << (can_reduce_string("aa")?"true":"false") << '\n';
    return 0;
}
